sklearn/ensemble/_stacking.py

```diff
@@ -256,6 +259,47 @@ def _transform(self, X):
         ]
         return self._concatenate_predictions(X, predictions)
 
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Input features.
+
+            - If `input_features` is `None`, then `feature_names_in_` is
+                used as feature names in. If `feature_names_in_` is not defined,
+                then names are generated: `[x0, x1, ..., x(n_features_in_)]`.
+            - If `input_features` is an array-like, then `input_features` must
+                match `feature_names_in_` if `feature_names_in_` is defined.
```

Comment:
Also, could you please make it explicit that input feature names impact the output features names only when `passthrough is True`? Otherwise `input_features` are not used to generate the output features of the ensemble and only the names of the base estimators in the ensemble are.

Reply:
> Also since it starts at x0, it should end at x(n_features_in_ - 1) in general for this kind of comments. I will do a dedicated PR.

Done in #22718.